src/qibolab/instruments/rfsoc.py

```diff
@@ -109,16 +125,33 @@ def convert_sweep(sweeper: Sweeper, sequence: PulseSequence, qubits: dict[int, Q
     else:
         for pulse in sweeper.pulses:
             indexes.append(sequence.index(pulse))
-
-            name = sweeper.parameter.name
-            parameters.append(getattr(rfsoc.Parameter, name.upper()))
-            base_value = getattr(pulse, name)
+            base_value = getattr(pulse, sweeper.parameter.name)
             values = sweeper.get_values(base_value)
             starts.append(values[0])
             stops.append(values[-1])
 
+            if sweeper.parameter not in {DURATION, START}:
+                parameters.append(convert_parameter(sweeper.parameter))
+            else:
+                if sweeper.parameter is DURATION:
+                    parameters.append(convert_parameter(sweeper.parameter))
+                elif sweeper.parameter is START:
+                    parameters.append(rfsoc.Parameter.START)
```

Comment:
```suggestion
                    parameters.append(convert_parameter(sweeper.parameter))
```
? 
Or is there a particular reason on why you coded in that way?

Reply:
It is fine like this, because it's a single value, so he can manually write the result (it's always the same, you can inline, mildly reducing computation, but denesting that for sure helps readability).

In principle, it could also be done in the case above, writing `rfsoc.Parameter.DURATION`. So, thanks in any case for pointing it out :)